app.py

```diff
@@ -0,0 +1,210 @@
+                value=False,
+            )
+            cond_free = st.checkbox(
+                "Conditioning Free", help="Force conditioning free diffusion", value=True
+            )
+            no_cond_free = st.checkbox(
+                "Force Not Conditioning Free",
+                help="Force disable conditioning free diffusion",
+                value=False,
+            )
+
+            """#### Debug"""
+            produce_debug_state = st.checkbox(
+                "Produce Debug State",
+                help="Whether or not to produce debug_state.pth, which can aid in reproducing problems. Defaults to true.",
+                value=True,
+            )
+
+    if 'tts' not in st.session_state:
+        st.session_state.tts = TextToSpeech(models_dir=model_dir, high_vram=high_vram, kv_cache=kv_cache)
+    tts = st.session_state.tts
+    if st.button("Start"):
+        with st.spinner(f"Generating {candidates} candidates for voice {voice} (seed={seed}). You can see progress in the terminal"):
+            os.makedirs(output_path, exist_ok=True)
+
+            selected_voices = voice.split(",")
+            for k, selected_voice in enumerate(selected_voices):
+                if "&" in selected_voice:
+                    voice_sel = selected_voice.split("&")
+                else:
+                    voice_sel = [selected_voice]
+                voice_samples, conditioning_latents = load_voices(voice_sel)
+
+                with timeit(
+                    f"Generating {candidates} candidates for voice {selected_voice} (seed={seed})"
+                ):
+                    nullable_kwargs = {
+                        k: v
+                        for k, v in zip(
+                            ["sampler", "diffusion_iterations", "cond_free"],
+                            [sampler, steps, cond_free],
+                        )
+                        if v is not None
+                    }
+                    gen, dbg_state = tts.tts_with_preset(
+                        text,
+                        k=candidates,
+                        voice_samples=voice_samples,
+                        conditioning_latents=conditioning_latents,
+                        preset=preset,
+                        use_deterministic_seed=seed,
+                        return_deterministic_state=True,
+                        cvvp_amount=0.0,
+                        half=half,
+                        **nullable_kwargs,
+                    )
+        if isinstance(gen, list):
+            for j, g in enumerate(gen):
+                filename = f"{selected_voice}_{k}_{j}.wav"
+                torchaudio.save(
+                    os.path.join(output_path, filename),
+                    g.squeeze(0).cpu(),
+                    24000,
+                )
+                audio_buffer = BytesIO()
+                torchaudio.save(
+                    audio_buffer,
+                    g.squeeze(0).cpu(),
+                    24000,
+                )
```

Comment:
bug: missed `format="wav"` (which is included in the copy of this line below); i'll edit this